CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -380,6 +380,7 @@ end
 -- Check if fatigue is over a certain level (decided by the hospital policy),
 -- and go to the StaffRoom if it is.
 function Staff:checkIfNeedRest()
+  if not self.attributes.fatigue then return end
```

Comment:
Have you sufficiently checked a fired staff member can still leave the room if their tiredness was over the need to rest threshold?

Reply:
Yes, they're on their way out of the world either way.